api/v1beta2/network_types.go

```diff
@@ -242,7 +242,7 @@ func (v *VPCSpec) IsIPv6Enabled() bool {
 // SubnetSpec configures an AWS Subnet.
 type SubnetSpec struct {
 	// ID defines a unique identifier to reference this resource.
-	ID string `json:"id,omitempty"`
+	ID string `json:"id"`
```

Comment:
So, what happens in case of existing clusters is that their ID will be set to a default empty string, right?
Does that affect the cluster at all?

Reply:
This means if a subnet spec is defined, it should have the id field.

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/6320

Reply:
Yeah, I saw this one. Just was wondering about existing configs since now it will not just omit the value but update it to an empty string.

Reply:
For existing configs (by this I mean existing clusters), even though initially id field was empty, after the creation, ids are filled by CAPA controllers, so no existing cluster should have their id fields empty during upgrading to v1beta2.

If you are asking about using an existing template, they will no longer work for creating new clusters. 
This is a breaking change for existing templates that use the unsupported use cases and the reason why we needed a new API version is because of this. We will need to document this properly with the v1beta2 release.